### PR TITLE
fix initialization

### PR DESCRIPTION
**Pull request by RomanRabirokh**
## Description (required)

corrects the initialization of the plugin when there are elements in the select before starting the search
and preserveSelected: true

## Screenshots (optional)

If this pull request affects any visual or UI aspects of this project
then you should place them here, preferably in a before/after format.

## Closes Issue (optional)
Closes  https://github.com/truckingsim/Ajax-Bootstrap-Select/issues/85

## Related Issue(s) (optional)

- X-ref https://github.com/truckingsim/Ajax-Bootstrap-Select/issues/152 

## Merge checklist (required, for maintainers only)

This pull request adheres to the following requirements:

- [] JavaScript follows 4 space indentation conventions.
- [] JavaScript was written using ES2015 (ES6) features.
- [] Features and bug fixes are covered by test cases.
- [] All commits follow the _AngularJS Git Commit Message Conventions_.

### CODE REVIEW

File: src/js/classes/AjaxBootstrapSelectList.js [hunk 1]
```diff
@@ -55,15 +56,27 @@ var AjaxBootstrapSelectList = function (plugin) {
             text: $option.text(),
             'class': $option.attr('class') || '',
             data: $option.data() || {},
-            preserved: plugin.options.preserveSelected,
```
Comment (truckingsim) [inline src/js/classes/AjaxBootstrapSelectList.js:59]: @RomanRabirokh why did you change this line? I think the rest of this makes sense for the most part, but not sure why you did this.

Comment (RomanRabirokh) [reply src/js/classes/AjaxBootstrapSelectList.js:59]: Oh. it was a long time ago. now hard to remember.

I have now tested, it does not affect anything.

But it is better to use your option. Because I do not exclude that this is my mistake.

**Comment by falegk** [comment]
These changes resolve the problem for me and looking forward to new version.
Also I think this line must stay as it is
`
preserved: plugin.options.preserveSelected
`

File: src/js/classes/AjaxBootstrapSelectList.js (hunk 1 above)
Comment (markhalliwell) [reply src/js/classes/AjaxBootstrapSelectList.js:59]: I agree, this shouldn't be changed.

**Comment by markcarver** [comment]
Please base PRs agains the `dev` branch.

**Review by markhalliwell — CHANGES_REQUESTED** [review]
Please base PRs agains the `dev` branch.

**Comment by RomanRabirokh** [comment]
@markcarver 
Oh. there are conflicts.

**Comment by markcarver** [comment]
Hm. Seem `dev` was severely outdated. It's fixed now.

**Comment by RomanRabirokh** [comment]
@markcarver 
i still have conflict.

I need to fix my  pool request
in this line (preserved: plugin.options.preserveSelected,)
 or will you fix it when merging?

**Comment by markcarver** [comment]
> or will you fix it when merging?

You'll need to fix it prior to merging this PR.

**Comment by RomanRabirokh** [comment]
I fixed and resolved conflicts

File: src/js/classes/AjaxBootstrapSelectList.js [hunk 2]
```diff
@@ -59,12 +60,23 @@ var AjaxBootstrapSelectList = function (plugin) {
...
+            selected: false
+        });
     });
-    this.cacheSet(/* query=*/'', initial_options);
+    this.cacheSet(/* query=*/'', initial_options_cache);
 
     // Preserve selected options.
     if (plugin.options.preserveSelected) {
         that.selected = initial_options;
+        if(that.selected) {
```
Comment (markhalliwell) [inline src/js/classes/AjaxBootstrapSelectList.js]: Needs space in between `if` and `(`.

File: src/js/classes/AjaxBootstrapSelectList.js [hunk 3]
```diff
@@ -282,7 +294,12 @@ AjaxBootstrapSelectList.prototype.replaceOptions = function (data) {
 
     // Build the option output.
     if (data.length) {
-        output = this.plugin.list.build(data);
+        if(this.plugin.list !== undefined) {
```
Comment (markhalliwell) [inline src/js/classes/AjaxBootstrapSelectList.js]: Needs space in between `if` and `(`.

**Review by markhalliwell — CHANGES_REQUESTED** [review]

**Review by markhalliwell — APPROVED** [review]
Thanks!
